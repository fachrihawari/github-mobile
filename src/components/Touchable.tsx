import React, { ReactNode, ReactType } from 'react';
import {
  Platform,
  TouchableNativeFeedback,
  TouchableOpacity
} from 'react-native';

interface ITouchableProps {
  onPress: () => void;
  children: ReactNode;
}

function Touchable({ children, onPress }: ITouchableProps) {
  const Component: ReactType =
    Platform.OS === 'ios' ? TouchableOpacity : TouchableNativeFeedback;

  return <Component testID="touchable" onPress={onPress}>{children}</Component>;
}

export default Touchable;
