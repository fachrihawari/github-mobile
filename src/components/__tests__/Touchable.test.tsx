import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';
import { Text, TouchableOpacity, Image, Platform, TouchableNativeFeedback } from 'react-native';

import Touchable from '../Touchable';

export const createTestProps = () => ({
  handlePress: jest.fn(),
  touchableLabel: 'Awesome Touchable',
  touchableImage: 'https://github.githubassets.com/images/modules/logos_page/GitHub-Mark.png'
})

describe('rendering', () => {
  const props = createTestProps()

  let wrapper: ShallowWrapper
  
  function newInstance() {
    wrapper = shallow(
      <Touchable testID='touchable' onPress={props.handlePress}>
        <Image source={{ uri: props.touchableImage }} />
        <Text>{props.touchableLabel}</Text>
      </Touchable>
    )
  }

  beforeEach(() => {
    newInstance()  
  })

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should render <TouchableOpacity/> for iOS', () => {
    Platform.OS = 'ios';
    newInstance()
    expect(wrapper.find({ testID: 'touchable' }).type() === TouchableOpacity).toBe(true)
  })

  it('should render <TouchableNativeFeedback/> for Android', () => {
    Platform.OS = 'android';
    newInstance()
    expect(wrapper.find({ testID: 'touchable' }).type() === TouchableNativeFeedback).toBe(true)
  })

  it('has pressed the touchable', () => {
    wrapper.simulate('press')
    expect(props.handlePress).toHaveBeenCalledTimes(1)
  })

})
