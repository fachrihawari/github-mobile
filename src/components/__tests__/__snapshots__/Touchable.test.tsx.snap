// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering should render correctly 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[MockFunction]}
  testID="touchable"
>
  <Image
    source={
      Object {
        "uri": "https://github.githubassets.com/images/modules/logos_page/GitHub-Mark.png",
      }
    }
  />
  <Text>
    Awesome Touchable
  </Text>
</TouchableOpacity>
`;
